lib/common/xml.c

```diff
@@ -1046,23 +1046,48 @@ pcmk__xml_needs_escape(const char *text, enum pcmk__xml_escape_type type)
             continue;
         }
 
-        switch (*text) {
-            case '<':
-                return true;
-            case '>':
-                return true;
-            case '&':
-                return true;
-            case '"':
-            case '\n':
-            case '\t':
-                if (type == pcmk__xml_escape_attr) {
-                    return true;
+        switch (type) {
+            case pcmk__xml_escape_text:
+                switch (*text) {
+                    case '<':
+                    case '>':
+                    case '&':
+                        return true;
+                    case '\n':
+                    case '\t':
+                        break;
+                    default:
+                        if (!isprint(*text)) {
+                            return true;
+                        }
+                        break;
                 }
                 break;
-            default:
-                if (!isprint(*text)) {
-                    return true;
+
+            case pcmk__xml_escape_attr:
+                switch (*text) {
+                    case '<':
+                    case '>':
+                    case '&':
+                    case '"':
+                        return true;
+                    default:
+                        if (!isprint(*text)) {
+                            return true;
+                        }
+                        break;
+                }
+                break;
+
+            case pcmk__xml_escape_attr_pretty:
+                switch (*text) {
+                    case '\n':
+                    case '\r':
+                    case '\t':
+                    case '"':
+                        return true;
+                    default:
+                        break;
```

Comment:
I'd rather not add another enum value for the sake of warning if we accidentally use it... I don't remember static analysis caring in that direction, but rather in the direction of wanting an exhaustive switch statement if there's no default case. Nothing complained in CI.

I'd rather `CRM_CHECK()` the type before the loop and return false. That approach would carry over to `pcmk__xml_escape()`: we certainly don't want to warn on every character that we append in the loop, and we can't return `pcmk__str_copy()` after we enter the loop because we might have skipped UTF-8 characters at the beginning.

Reply:
or `CRM_ASSERT()` since it's purely internal and that's definitely a programming error on our side

Reply:
That would be fine. The issue is that enum variables are just integers and not limited to specified values. So if you want to use a default case to check for unspecified values, static analysis will complain that no values can match the default. Adding an unknown/invalid value and letting default catch it along with unspecified values is a good way around that.

I assume you're thinking of asserting that the value is >= the first value and <= the last. That should work too. I have seen a corner case where the value was unsigned for some reason and testing >= 0 got a complaint that it would always be true.

Reply:
I did it even more simply (see current commits):
```
default:
    CRM_ASSERT(false);
    break;
```

Neither Coverity nor cppcheck complained about it